src/ui-components/FormFields/Autocomplete.tsx

```diff
@@ -28,25 +29,35 @@ const AutocompleteField = <T extends IAutoCompleteItem>({
   id,
   selectedItem,
   resetSelectedItem,
+  scrollToError,
 }: IAutocompleteProps<T>) => {
   const { control } = useFormContext()
   const { field, fieldState } = useController({ name: fieldName, control })
   const errorMessage = error ?? fieldState?.error?.message
+  const scrollRef = useRef<HTMLDivElement | null>(null)
+
+  useEffect(() => {
+    if (scrollToError && fieldState.error && scrollRef.current) {
+      scrollRef.current?.scrollIntoView({ behavior: 'smooth', block: 'end', inline: 'nearest' })
+    }
+  }, [fieldState, fieldState.error, scrollToError])
```

Comment:
Can we make this a hook?

Reply:
yes we can, it's done waiting for your review.